state-chain/amm-math/src/lib.rs

```diff
@@ -108,6 +109,27 @@ pub fn output_amount_ceil(input: Amount, price: Price) -> Amount {
 	mul_div_ceil(input, price, U256::one() << PRICE_FRACTIONAL_BITS)
 }
 
+pub fn price_from_usd_fine_amount(price_usd: AssetAmount) -> Price {
+	Price::from(price_usd) << PRICE_FRACTIONAL_BITS
+}
+
+/// Get the price of an asset given its dollar amount and decimals.
+/// Using u32 for dollar_amount so it is type safe against fine amounts (u128).
+pub fn price_from_usd(dollar_amount: u32, decimals: u32) -> Price {
```

Comment:
We could make this `price_from_usd(asset: impl Into<Asset>, dollar_amount: u32)` and then the caller doesn't need to worry about the correct number of decimal places? 

Also, am I correct that this only accepts whole dollar amounts? What if we make it cents?